src/elf/sym.rs

```diff
@@ -320,112 +318,111 @@ pub mod sym64 {
+    #[inline]
+    pub fn st_bind(&self) -> u8 {
+        self.st_info >> 4
+    }
+    /// Get the ST type.
+    ///
+    /// This is the last four bits of the "info" byte.
+    #[inline]
+    pub fn st_type(&self) -> u8 {
+        st_type(self.st_info)
+    }
+    /// Get the ST visibility.
+    ///
+    /// This is the last three bits of the "other" byte.
+    #[inline]
+    pub fn st_visibility(&self) -> u8 {
+        st_visibility(self.st_other)
+    }
+    #[cfg(feature = "endian_fd")]
+    /// Parse `count` vector of ELF symbols from `offset`
+    pub fn parse(bytes: &[u8], mut offset: usize, count: usize, ctx: Ctx) -> Result<Vec<Sym>> {
+        let mut syms = Vec::with_capacity(count);
+        for _ in 0..count {
+            let sym = bytes.gread_with(&mut offset, ctx)?;
+            syms.push(sym);
         }
+        Ok(syms)
     }
+}
 
-    impl fmt::Debug for Sym {
-        fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
-            let bind = self.st_bind();
-            let typ = self.st_type();
-            let vis = self.st_visibility();
-            f.debug_struct("Sym")
-                .field("st_name", &self.st_name)
-                .field("st_info", &format_args!("0x{:x} {} {}", self.st_info, bind_to_str(bind), type_to_str(typ)))
-                .field("st_other", &format_args!("{} {}", self.st_other, visibility_to_str(vis)))
-                .field("st_shndx", &self.st_shndx)
-                .field("st_value", &format_args!("0x{:x}", self.st_value))
-                .field("st_size", &self.st_size)
-                .finish()
-        }
+impl fmt::Debug for Sym {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        let bind = self.st_bind();
+        let typ = self.st_type();
+        let vis = self.st_visibility();
+        f.debug_struct("Sym")
+            .field("st_name", &self.st_name)
+            .field("st_info", &format_args!("0x{:x} {} {}", self.st_info, bind_to_str(bind), type_to_str(typ)))
+            .field("st_other", &format_args!("{} {}", self.st_other, visibility_to_str(vis)))
+            .field("st_shndx", &self.st_shndx)
+            .field("st_value", &format_args!("0x{:x}", self.st_value))
+            .field("st_size", &self.st_size)
+            .finish()
     }
+}
 
-    impl ctx::SizeWith<Ctx> for Sym {
-        #[inline]
-        fn size_with(&Ctx {container, .. }: &Ctx) -> usize {
-            match container {
-                Container::Little => {
-                    sym32::SIZEOF_SYM
-                },
-                Container::Big => {
-                    sym64::SIZEOF_SYM
-                },
-            }
+impl ctx::SizeWith<Ctx> for Sym {
+    #[inline]
+    fn size_with(&Ctx {container, .. }: &Ctx) -> usize {
+        match container {
+            Container::Little => {
+                sym32::SIZEOF_SYM
+            },
+            Container::Big => {
+                sym64::SIZEOF_SYM
+            },
         }
     }
+}
 
+if_alloc! {
```

Comment:
How come the TryFromCtx impls require alloc ?

Reply:
The error::Error has variant `Malformed(String)` which requires allocation.

Reply:
Ah yea. Hmmm, might be worth investigating whether static strs could replace or more variants. Anyway thanks for PR!